hub/api/dataset.py

```diff
@@ -75,11 +76,7 @@ def __init__(
             creds = {}
         base_storage = get_storage_provider(path, storage, read_only, creds, token)
 
-        # done instead of directly assigning read_only as backend might return read_only permissions
-        if hasattr(base_storage, "read_only") and base_storage.read_only:
-            self._read_only = True
-        else:
-            self._read_only = False
+        self._read_only = base_storage.read_only
```

Comment:
so is read-only properly set if backend says the user has no permissions to write?

Reply:
not really, the user passed flag is also passed to base_storage in get_storage_provider. So either way base_storage will reflect the state.